File: cpe/cpp/zj-e629.cpp

#include <iostream>
#include <cmath>
using namespace std;

int main() {
    int S, Case = 1;
    while (cin >> S){
        if (S == 0) break;
        cout << "Case " << Case++ << ": ";
        if (S == 1){
            cout << "1\n";
            continue;
        }
        int ans = -1;
        for (int i = S-1; i >= 1; i--){
            int sum = 0;
            for (int j = 1; j <= sqrt(i); j++){
                if (i % j == 0){
                    sum += j;
                    if (j * j != i) sum+= i/j;
                }
            }
            if (sum == S){
                ans = i;
                break;
            }
        }
        cout << ans << "\n";
    }
    return 0;
}

File: cpe/python/zj-e629.py
#!/usr/bin/env python 

Case = 1
while True:
    S = int(input())
    if S == 0:
        break
    print(f"Case {Case}: ", end="")
    Case += 1
    if S == 1:
        print(1)
        continue
    ans = -1
    for i in range(S-1, 0, -1):
        Sum = 0
        for j in range(1, int(i**0.5)+1):
            if i % j == 0:
                Sum += j
                if j*j != i:
                    Sum += i // j

        if Sum == S:
            ans = i
            break
    print(ans)
